src/command_modules/azure-cli-webapp/azure/cli/command_modules/webapp/_params.py

```diff
@@ -7,52 +7,60 @@
 from azure.mgmt.web import WebSiteManagementClient
 from azure.cli.core.commands.client_factory import get_mgmt_service_client
 from azure.cli.core.commands.parameters import (resource_group_name_type, location_type,
+                                                get_resource_name_completion_list,
                                                 CliArgumentType, ignore_type)
 
 # FACTORIES
 def web_client_factory(**_):
     return get_mgmt_service_client(WebSiteManagementClient)
 
-def list_webapp_names(prefix, action, parsed_args, **kwargs):#pylint: disable=unused-argument
-    if parsed_args.resource_group_name:
-        client = web_client_factory()
-        #pylint: disable=no-member
-        result = client.sites.get_sites(parsed_args.resource_group_name).value
-        return [x.name for x in result]
+def _generic_site_operation(resource_group_name, name, operation_name, slot=None, #pylint: disable=too-many-arguments
+                            extra_parameter=None, client=None):
+    client = client or web_client_factory()
+    m = getattr(client.sites,
+                operation_name if slot is None else operation_name + '_slot')
+    if slot is None:
+        return (m(resource_group_name, name)
+                if extra_parameter is None else m(resource_group_name, name, extra_parameter))
+    else:
+        return (m(resource_group_name, name, slot)
+                if extra_parameter is None else m(resource_group_name, name, extra_parameter, slot))
+
+def get_hostname_completion_list():
+    def completer(prefix, action, parsed_args, **kwargs): # pylint: disable=unused-argument
```

Comment:
You shouldn't need the wrapper function.
The below should work fine as no params are being passed into `get_hostname_completion_list` currently.
```
def get_hostname_completion_list(prefix, action, parsed_args, **kwargs):
    ...
```

Then, when you register the CLI argument, it would be `(completer=get_hostname_completion_list)`.